source/module_base/test/complexarray_test.cpp

```diff
@@ -45,125 +50,417 @@
 *   - operator "*=":
 *       similar as "*"
 *
+*   - operator "==":
+*       judge if two ComplexArray is equal, all element is equal.
+*
+*   - operator "!=":
+*       judge if two ComplexArray is not equal
+*
 *   - oprator "()":
 *       - access the element
+*
+*   - function zero_out, negate, randomize, getBound/Size, create.
+*
+*   Test relative functions:
+*   - overloading of operator "*". a double/complex multiply a Class ComplexArray.
+*   - functon abs2(), return the sum of squares of all elements 
+*   - function dot()
+*   - function scale_accumulate()
+*   - function scaled_sum()
+*   - fucntion point_mult()
 */
 
 
+double prec_test = 1e-14;
+//compare two complex by using EXPECT_NEAR()
+void EXPECT_COMPLEXD(const std::complex<double>& a,const std::complex<double>& b)
+{
+    EXPECT_NEAR(a.real(),b.real(),prec_test);
```

Comment:
Shall we use `EXPECT_DOUBLE_EQ` when comparing? What's the point of designate a precision threshold?